python/cudf/cudf/dataframe/dataframe.py

```diff
@@ -1062,8 +1062,23 @@ def take(self, positions, ignore_index=False):
                 out.index = RangeIndex(len(out))
             else:
                 out.index = self.index.take(positions)
+
+        if isinstance(self.index, cudf.MultiIndex):
+            out._index = self.index.take(positions)
```

Comment:
So moving this out of the `if` block actually has correct behavior for MultiIndex if there's no columns, but normal indexes are incorrect. Could we just clean up this block to fix the behavior of both and align both to use `out.index = self.index.take(positions)`? Reproducer:
```python
import cudf
gdf = cudf.DataFrame(None, index=[0,1,2,3,4])
gdf.take([0,2])
```

Reply:
I addressed your concern, but in order to pass all tests I needed conditional logic inside of `dataframe.py::take` to decide how to set the new index.